accumulator_updater/programs/mock-cpi-caller/src/instructions/cpi_max_test.rs

```diff
@@ -0,0 +1,54 @@
+use {
+    crate::{
+        instructions::{
+            UpdatePrice,
+            UpdatePriceParams,
+        },
+        message::{
+            get_schemas,
+            price::{
+                CompactPriceMessage,
+                DummyPriceMessage,
+                FullPriceMessage,
+            },
+            AccumulatorSerializer,
+        },
+        state::PythAccountType,
+    },
+    anchor_lang::prelude::*,
+};
+
+pub fn cpi_max_test<'info>(
+    ctx: Context<'_, '_, '_, 'info, UpdatePrice<'info>>,
+    params: UpdatePriceParams,
+    num_messages: u8,
+) -> Result<()> {
+    let mut inputs = vec![];
+    let _schemas = get_schemas(PythAccountType::Price);
+
+    {
+        let pyth_price_acct = &mut ctx.accounts.pyth_price_account.load_mut()?;
+        pyth_price_acct.update(params)?;
+
+        let price_full_data = FullPriceMessage::from(&**pyth_price_acct).accumulator_serialize()?;
+
+        inputs.push(price_full_data);
+
+
+        let price_compact_data =
+            CompactPriceMessage::from(&**pyth_price_acct).accumulator_serialize()?;
+        inputs.push(price_compact_data);
```

Comment:
fwiw i think this would be a cleaner test if this pushed only the dummy messages. You really want to be able to precisely control the amount of data being written into the accumulator, and having multiple message types makes it harder to do that.

In fact, I think the optimal way to write this test would have been to like write an array of bytes where the caller chooses the length of the array.

(not a big deal though)

Reply:
updated test following your suggestion